bin/daemon

```diff
@@ -29,17 +29,26 @@ end
 @checksum = calc_checksum
 
 module Updater
+  class UpdateFailed < StandardError
+  end
+
   extend Scarlet::Daemon::Loggable
 
   def self.git_update
     data = Scarlet::Git.get_data
     branch = data[:branch]
-    throw :update_failed, 'no branch' unless branch
-    system("git fetch origin") || throw(:update_failed, 'Fetching failed')
-    system("git reset --hard origin/#{branch}") || throw(:update_failed, 'Reset failed')
+    raise UpdateFailed, 'no branch obtained.' unless branch
+
+    system("git fetch origin") || raise(UpdateFailed, 'git fetch failed')
+    system("git reset --hard origin/#{branch}") || raise(UpdateFailed, 'git reset failed')
+
     current_data = Scarlet::Git.get_data
     log "Updated from #{data[:commit]} to #{current_data[:commit]}"
-    false # false as in, did I fail?
+  end
+
+  def self.install_gems
+    log "Installing gems"
+    system("bundle install")
   end
 
   def self.git_rollback(commit, reason)
```

Comment:
That would create a detached HEAD, and prevents us from updating a specific branch.